### ISSUE TITLE
Bug: Slide Box with 2 slides and continuous play adds 2 cloned slides (for no reason?)

### ISSUE THREAD

**Issue by bra1n** [driftyco/ionic#2336]
Hi,
I have the following situation: I'm using a Slide Box that gets its elements from an asynchronous AJAX call and then populates the Slide Box container with ng-repeated slides like this:
```
          <ion-slide-box show-pager="false" does-continue="true" auto-play="true">
            <ion-slide ng-repeat="app in data.apps">
              <directive-app app="app"></directive-app>
            </ion-slide>
          </ion-slide-box>
```
This works fine as long as there are either 1 or 3+ elements in my data.apps array. However, when there are exactly 2 elements, the following code from `sliderView.js` kicks in:
``` 
      //special case if two slides
      if (browser.transitions && options.continuous && slides.length < 3) {
        element.appendChild(slides[0].cloneNode(true));
        element.appendChild(element.children[1].cloneNode(true));
        slides = element.children;
      }
```
This causes the browser to create 2 empty `directive-app` elements inside 2 new cloned slides, since the directives won't be parsed by AngularJS in cloned elements. When I removed the code block that clones these elements, everything worked fine again, including continuous looping and auto-play. Now I wonder, what exactly is the reason to create a copy of both slides? Especially when it is not recommended to  clone HTML nodes that have Angular logic attached through plain JS or similar means and it works just fine without the cloned slides. 

In any case, it looks like when this code was created that it didn't account for directives or other sub-templates inside the slides and I suggest to just remove the whole block, as it seems to work perfectly fine without the clones.

Thanks,
Steffen

**Comment by VincentPuget** [comment]
+1

**Comment by Benew** [comment]
+1000 ;-)

**Comment by NBAMj** [comment]
+1
I'm still facing this issue in Ionic#1.1.0.

**Comment by Clawg** [comment]
+1 same for me on 1.1.0

**Comment by mlynch** [comment]
Hey everyone, can you try the new slidebox just pushed to master (available in upcoming 1.2):

```html
<ion-slides options="options">
  <ion-slide-page>
  </ion-slide-page>
</ion-slides>
```

Handles the < 3 slide and looping case a lot better! Specify options in the options scope variable:

```javascript
controller('MyCtrl', function($scope) {
  $scope.options = {
    loop: true
  };
});
```

**Comment by made21** [comment]
im on meteor 1.5 and ionic and so far still have the issue
